src/tbx/mod.rs

```diff
@@ -0,0 +1,410 @@
+// Copyright 2018 Manuel Holtgrewe, Berlin Institute of Health.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::ffi;
+use std::mem;
+use std::ptr;
+use std::path::Path;
+use url::Url;
+use libc;
+
+use htslib;
+
+/// A trait for a Tabix reader with a read method.
+pub trait Read: Sized {
+    /// Read next line the given `String`.
+    /// Use this method in combination with a single allocated record to avoid the reallocations
+    /// occurring with the iterator.
+    ///
+    /// # Arguments
+    ///
+    /// * `record` - the `String` to be filled
+    fn read(&mut self, record: &mut String) -> Result<(), ReadError>;
+
+    /// Iterator over the lines/records of the seeked region.
+    /// Note that, while being convenient, this is less efficient than pre-allocating a
+    /// `String` and reading into it with the `read` method, since every iteration involves
+    /// the allocation of a new `String`.
+    fn records(&mut self) -> Records<Self>;
+
+    /// Return the text headers, split by line.
+    fn header(&self) -> &Vec<String>;
+}
+
+/// A Tabix file reader.
+#[derive(Debug)]
+pub struct TabixReader {
```

Comment:
Should this be called `IndexedReader`, similar to `bam::IndexedReader`?

Reply:
Hm, I'm not sure. What is the `Reader` then? Simply something that does `gunzip`? Please you decide, I have no strong opinion.

Reply:
Ok, then I decide for aehm, `Reader`.

Reply:
OK, pushed the change.